app/subscriber/src/features/settings/MyAccountSettings.tsx

```diff
@@ -48,6 +51,40 @@ const MyAccountSettings = () => {
     [updateUser],
   );
 
+  const toggleReportSentiment = React.useCallback(
+    async (
+      profile: ISubscriberUserModel | undefined,
+      impersonate: ISubscriberUserModel | undefined,
+      enableReportSentiment: boolean,
+    ) => {
+      if (!profile) {
+        toast.error('User information is missing. Please try again later');
+        return;
+      }
+      const baseProfile = impersonate ?? profile;
+      const createUser = (): ISubscriberUserModel => {
+        // use impersonate if it exists, otherwise use profile
+        return {
+          ...baseProfile,
+          preferences: {
+            ...baseProfile.preferences,
+            enableReportSentiment: enableReportSentiment,
+          },
+        };
+      };
+      const user = createUser();
+
+      try {
+        await updateUser(user, !!impersonate);
+        toast.success('Report sentiment has successfully been updated.');
```

Comment:
It's not really related to reports presently.  This change appears to only impact notifications/alerts.